winter/http/response_header.py

```diff
@@ -0,0 +1,53 @@
+from typing import MutableMapping
+from typing import Type
+from typing import TypeVar
+
+import dataclasses
+
+from winter.core import annotate_method
+from winter.converters import convert
+
+T = TypeVar('T')
+
+
+class _BaseResponseHeader:
+    _value_type: Type[T] = None
+
+    def __init__(self, headers: MutableMapping[str, str], header_name: str):
+        self._headers = headers
+        self._header_name = header_name
+
+    def set(self, value: T):
+        self._headers[self._header_name] = convert(value, self._value_type)
+
+
+class _ResponseHeader:
+    def __getitem__(self, value_type: Type):
+        assert isinstance(value_type, type), 'value_type must be a type'
+        payload = {
+            '_value_type': value_type,
+        }
+        return type(f'ResponseHeader[{value_type.__name__}]', (_BaseResponseHeader,), payload)
```

Comment:
typing.Dict.__name__  raises AttributeError: __name__

Reply:
should we leave ResponseHeader[] for example?

Reply:
I suggest f'ResponseHeader[{value_type}]'

Reply:
typing.Dict[str, str].__name__ is Dict, and str(typing.Dict[str, str].__name__) is typing.Dict[str, str]